sandbox/apparmor/notify/listener/listener.go

```diff
@@ -0,0 +1,251 @@
+// Package listener implements a high-level interface to the apparmor
+// notification mechanism. It can be used to build userspace applications
+// which respond to apparmor prompting profiles.
+package listener
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"gopkg.in/tomb.v2"
+
+	"github.com/snapcore/snapd/osutil/epoll"
+	"github.com/snapcore/snapd/sandbox/apparmor/notify"
+)
+
+// Listener contains low-level components for receiving notification requests
+// and responding with notification responses.
+type Listener struct {
+	// R is a channel with incoming requests. Each request is asynchronous
+	// and needs to be replied to.
+	R chan *Request
+	// E is a channel for receiving asynchronous error messages from
+	// concurrently running parts of the listener system.
+	E chan error
```

Comment:
the std lib uses this kind of very shorts names for channels but I'm not sure they are great outside of it, I think Reqs and Errs would be clearer names here. Also if it's not too annoying it would be good to use directional channel types for the exported channels.